build.sh

```diff
@@ -168,6 +170,7 @@ fi
 make iso BASE="${MYBASE}" RELEASE="${MYRELEASE}" ARCH="${MYARCH}" ROOTPW_HASH="*"
 
 # scp to distribution site
+# shellcheck disable=SC2154 # this relates to above, can ignore?
```

Comment:
I would suggest to remove the `shellcheck disable line` here and solve the underlying issues:

1. MYREMOTECONFIG should be set to `settings.sh` instead of settings.cfg to make it clear that it's sourcing a script (not strictly needed, but nicer for the user as it makes things clear)
2.  -u be turned from an option with arguments to a flag (so one would say `./build.sh -u` to upload). Basically:
```
  u) UPLOAD="YES" ;;
  ...
   if [ "$UPLOAD" = "YES" ]; then
   ...
   fi
```
3. All things set by settings.sh should be initialized to empty or good default values(e.g., a good default value for CFG_REMOTEPORT is empty, so that what is configured for the host in .ssh/config catches on) before importing them. I would still use UPPERCASE for all settings, as they're shell variables that are pulled in - maybe prefix them with `CFG_`
4. Within the "if UPLOAD" block, required parameters should be checked.

What I would suggest is to only allow setting:
- CFG_SSH_REMOTEHOST
- CFG_SSH_REMOTEPATH (default empty)

and leave all remaining up to .ssh/config.

If you allow overriding more values (for reasons), make sure they're not used at all if they're not set (bash makes this really easy). This will make sure that entries in ~/.ssh/config are used unless explicitly overridden by the user. E.g.

```
ssh_args=()
if [ -n "$CFG_SSH_REMOTEPORT" ]; then
  ssh_args+=(-p "$CFG_SSH_REMOTEPORT")
fi
# ...more assembly of options...
scp "${ssh_args[@]}" 
```

I can help you with scripting details if you get stuck.

Reply:
I have a workable solution here